### PR TITLE
MRG, DOC: Update contrib guide

### LINKED ISSUE #6898: DOC: improvements to contributing guide
This issue is to collect suggestions from Woods Hole code sprint participants about how to improve the [contributing page](https://mne.tools/stable/install/contributing.html). Add your comments below!

### PR DESCRIPTION
**Pull request by drammock**
closes #6898 

This PR also updates the FAQ on filing new bugs, and the bug report issue template.

cc @mohdsherif @jasmainak @massich

### CODE REVIEW

**Comment by jasmainak** [comment]
Perhaps you can say "origin" next to "your fork" like you do for upstream?

**Comment by jasmainak** [comment]
maybe @AdoNunes can help review this. He has been reading these instructions recently

**Comment by massich** [comment]
I like it.

File: doc/_static/diagrams/git_setup.dot [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    splines=true
+    fontsize=20
+    fontname="Source Sans Pro"
+
+    node [shape=box
+          style=filled
+          fontsize=12
+          fontname="Source Sans Pro"
+          fillcolor="#bbbbbb"
+          margin="0.35,0.1"]
```
Comment (larsoner) [inline doc/_static/diagrams/git_setup.dot]: Do you need/want links anywhere? If so you can get them in Graphviz, for example for nodes or edges you can do (in Python syntax here):
```
            edge.attr['URL'] = '../overview.html'
            edge.attr['target'] = '_top'
```
You need to render in SVG mode. The `..` in the link is necessary to get out of the SVG itself.

Comment (drammock) [reply doc/_static/diagrams/git_setup.dot]: I'm not sure there's a lot of value in creating links from this diagram. The `git clone` and `git remote add` commands are shown in the very next section, and there's no way to link to an unknown user's fork or to their local computer, so the only link that makes sense is the first node (the main MNE-Python repo).

**Comment by drammock** [comment]
marking as MRG, but I'm fine with waiting for additional reviews if anyone else wants to chime in

**Comment by agramfort** [comment]
@larsoner merge if happy

**Comment by larsoner** [comment]
Thanks @drammock

**Pull request merged**